src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -92,8 +79,21 @@ private void AppendIntersectingSpansInSortedOrder(SnapshotSpan snapshotSpan, Seg
             result.Add(GetTranslatedTagSpan(tagSpan, snapshot, _spanTrackingMode));
     }
 
-    public IEnumerable<TagSpan<TTag>> GetSpans(ITextSnapshot snapshot)
-        => _tree.Select(tn => GetTranslatedITagSpan(tn, snapshot));
+    /// <summary>
+    /// Gets all the tag spans in this tree, remapped to <paramref name="snapshot"/>, and returns them as a <see
+    /// cref="NormalizedSnapshotSpanCollection"/>.
+    /// </summary>
+    public NormalizedSnapshotSpanCollection GetSnapshotSpanCollection(ITextSnapshot snapshot)
+    {
+        using var _ = ArrayBuilder<SnapshotSpan>.GetInstance(out var spans);
+
+        foreach (var tagSpan in _tree)
+            spans.Add(GetTranslatedSpan(tagSpan, snapshot, _spanTrackingMode));
+
+        return spans.Count == 0
+            ? NormalizedSnapshotSpanCollection.Empty
+            : new(spans);
```

Comment:
> new(spans); [](http://example.com/codeflow?start=13&length=12)

nit: this ends up calling the IEnumerable ctor, right?

Reply:
yup!  is tehre an alternative?

Reply:
I guess not. There is an IList ctor for NormalizedSnaphotSpanCollection which doesn't use an enumerator, but it doesn't look like ArrayBuilder implements IList<T>?

Reply:
yeah, i can live with this for now :)